runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkProcessContext.java

```diff
@@ -125,8 +164,50 @@ public InputT element() {
 
   @Override
   public void outputWithTimestamp(OutputT output, Instant timestamp) {
-    output(WindowedValue.of(output, timestamp,
-        windowedValue.getWindows(), windowedValue.getPane()));
+    if (windowedValue == null) {
+      // this is start/finishBundle.
+      output(noElementWindowedValue(output, timestamp, windowingStrategy));
+    } else {
+      output(WindowedValue.of(output, timestamp, windowedValue.getWindows(),
+          windowedValue.getPane()));
+    }
+  }
+
+  static <T> WindowedValue<T> noElementWindowedValue(final T output,
+                                                     final Instant timestamp,
+                                                     WindowingStrategy<?, ?> ws) {
+    @SuppressWarnings("unchecked")
+    WindowFn<Object, ?> windowFn = (WindowFn<Object, ?>) ws.getWindowFn();
+    WindowFn.AssignContext assignContext = windowFn.new AssignContext() {
+
+      @Override
+      public Object element() {
+        return output;
+      }
+
+      @Override
+      public Instant timestamp() {
+        if (timestamp != null) {
+          return timestamp;
+        }
+        throw new UnsupportedOperationException("outputWithTimestamp was called with "
+            + "null timestamp.");
+      }
+
+      @Override
+      public BoundedWindow window() {
+        throw new UnsupportedOperationException("Window not available for "
+            + "start/finishBundle output.");
+      }
+    };
+    try {
+      @SuppressWarnings("unchecked")
+      Collection<? extends BoundedWindow> windows = windowFn.assignWindows(assignContext);
+      Instant outputTimestamp = timestamp != null ? timestamp : BoundedWindow.TIMESTAMP_MIN_VALUE;
```

Comment:
I don't believe `timestamp` can ever be null, given that this is called from `outputWithTimestamp` and variants.